### PR TITLE
Improve the CSP compatibility of SimpleCov (Rebase #51)

### PR DESCRIPTION
**Pull request by PragTob**
* update libraries, rebuild assets
* enable sprockets js & css compression
* use $.fn.on() instead of $.fn.live() which is removed
* replace inline style with class
* remove url plugin
* use https

Rebase of #51 - will try to update all libraries again here and fix remaining issues. Stay tuned.

### CODE REVIEW

**Comment by PragTob** [comment]
Amazingly this seems to work (mostly), even the data table sorting.

CSS is broken though atm:

Old:

image

Current:

Selection_103

(not that I'd wanna keep the old layout, but some things are clearly broken)

**Comment by PragTob** [comment]
Minor tweaking and CSS inclusion later:

image

**Comment by PragTob** [comment]
Fixes: https://github.com/colszowka/simplecov/issues/849

**Comment by PragTob** [comment]
Ah, now I found the bug with sorting again :(

Fixing...

File: assets/javascripts/application.js [hunk 1]
```diff
@@ -5,20 +5,8 @@
...
-  lineColumns = [null, { "sType": "percent" }, null, null, null, null, null];
-  branchColumns = [{ "sType": "percent" }, null, null, null];
-
-  dataColumns = branchCoverageActivated ? lineColumns.concat(branchColumns) : lineColumns;
-
   $('.file_list').dataTable({
-    "aaSorting": [[1, "asc"]],
-    "bPaginate": false,
-    "bJQueryUI": true,
-    "aoColumns": dataColumns
```
Comment (klyonrad) [inline assets/javascripts/application.js]: Awesome simplification!

(datatables is my trauma and pet peeve

File: assets/javascripts/application.js [hunk 2]
```diff
@@ -5,20 +5,8 @@
...
+    "aaSorting": [[1, "asc"]]
```
Comment (klyonrad) [inline assets/javascripts/application.js]: Are you sure that this is still a valid option in the API? I can only find it in the legacy docs https://legacy.datatables.net/ref#aaSorting

Comment (PragTob) [reply assets/javascripts/application.js]: :wave: 

I'm not. However it still sorted the column I wanted to ascending so I was like :man_shrugging: "maybe a hidden compatibility layer"

I can try removing it, maybe it's another "default magic" thing from datatables

**Pull request merged**